test/e2e/common_test.go

```diff
@@ -40,9 +45,32 @@ func defaultShootCreationFramework() *framework.ShootCreationFramework {
 	})
 }
 
-func defaultShoot(generateName string, blackholing bool, blockAddress string) *gardencorev1beta1.Shoot {
+func defaultShoot(generateName string, blackholing bool, blockAddress string, useWgBlackholing, wgBlackholing bool, groups []string) *gardencorev1beta1.Shoot {
```

Comment:
Would it not be better to use just one parameter with type `*WorkerSpecific` and unconditionally set it? Callers would only need to provide one parameter if unused (`nil`) or specify the struct directly.

WDYT?

Reply:
I think the original intent for this function was to declare a shoot with a single line of code in the test cases and avoid building up structs outside. When I pass a `*WorkerSpecific` pointer I have to pre-fill it somewhere else, moving the complexity out of the func and inside the test case which kind of defeats the purpose of the func.

I do agree it's a little lot of arguments now, though. We could split it into two funcs and save one param:
```
func defaultShoot(generateName string, blackholing bool, blockAddress string)

func workerSpecificShoot(generateName string, blackholing bool, blockAddress string, wgBlackholing bool, groups []string)
```

we could drop `useWgBlackholing` as it is implicitly true in `workerSpecificShoot`